### PR TITLE
Add help target to list all different targets

### PR DESCRIPTION
**Pull request by nanikjava**
Fix: #5603

Excuting 'make help' will display a lit of available targets
for building minikube. The output will look as follows
```

Available targets for minikube v1.4.1
--------------------------------------
iso-menuconfig                 Configure buildroot configuration
linux-menuconfig               Configure Linux kernel configuration
all                            Build all different minikube component
drivers                        Build Hyperkit and KVM2 drivers
docker-machine-driver-hyperkit Build Hyperkit driver
integration                    Trigger minikube integration test
integration-none-driver        Trigger minikube none driver test
integration-versioned          Trigger minikube integration testing
```

...
...
...

### CODE REVIEW

**Comment by tstromberg** [comment]
Seems OK, but will defer judgement to the resident `Makefile` expert, @afbjorklund.

File: Makefile [hunk 1]
```diff
@@ -289,34 +289,34 @@ checksum:
...
 	rm -f pkg/minikube/translate/translations.go
 	rm -rf ./vendor
 
 .PHONY: gendocs
-gendocs: out/docs/minikube.md
+gendocs: out/docs/minikube.md  ## Generate documentation
 
 .PHONY: fmt
-fmt:
+fmt: ## Run fmt
```
Comment (afbjorklund) [inline Makefile]: Maybe: "Run go fmt and modify files in place" ?

File: Makefile [hunk 2]
```diff
@@ -289,34 +289,34 @@ checksum:
...
 	@go vet $(SOURCE_PACKAGES)
 
 .PHONY: golint
-golint:
+golint: ## Run golint
 	@golint -set_exit_status $(SOURCE_PACKAGES)
 
 .PHONY: gocyclo
-gocyclo:
+gocyclo: ## Run gocyclo (calculates cyclomatic complexities of functions in Minikube source code)
```
Comment (afbjorklund) [inline Makefile]: too long, better with something like "Run gocyclo (calculates cyclomatic complexities)"

File: Makefile [hunk 3]
```diff
@@ -525,29 +525,34 @@ kvm_in_docker:
...
+site: site/themes/docsy/assets/vendor/bootstrap/package.js out/hugo/hugo ## Serve the documentation site to localhost
 	(cd site && ../out/hugo/hugo serve \
 	  --disableFastRender \
 	  --navigateToChanged \
 	  --ignoreCache \
 	  --buildFuture)
+
+.PHONY: help
+help:
+	@echo "\e[1mAvailable targets for minikube ${VERSION}\e[21m"
```
Comment (afbjorklund) [inline Makefile]: These don't seem to be working here:

```console
$ make help
\e[1mAvailable targets for minikube v1.4.1\e[21m
\e[1m--------------------------------------\e[21m
...
```

The `printf` version works (listed in cyan), so maybe use that (for bold) instead ?

```sh
printf "\033[1mAvailable targets for minikube ${VERSION}\033[21m\n"
```

**Comment by afbjorklund** [comment]
@nanikjava : please rebase/squash into one single commit, and fix the commit message:

You need to remove the `Fix:`, and there is a typo in `display a lit of available targets`

**Comment by nanikjava** [comment]
> @nanikjava : please rebase/squash into one single commit, and fix the commit message:
> 
> You need to remove the `Fix:`, and there is a typo in `display a lit of available targets`

Done. Thank you.

**Comment by afbjorklund** [comment]
Can you fix the bold as well ? Thanks.

```
\e[1mAvailable targets for minikube v1.4.1\e[21m
\e[1m--------------------------------------\e[21m
```

**Comment by nanikjava** [comment]
> Can you fix the bold as well ? Thanks.
> 
> ```
> \e[1mAvailable targets for minikube v1.4.1\e[21m
> \e[1m--------------------------------------\e[21m
> ```

Sorry you mean remove the bold ?

**Comment by afbjorklund** [comment]
> Sorry you mean remove the bold ?

I meant to use the same printf function as in the actual help:

```Makefile
help:
	@printf "\033[1mAvailable targets for minikube ${VERSION}\033[21m\n"
	@printf "\033[1m-------------------------------------\033[21m\n"
	@grep -h -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'

```

**Comment by nanikjava** [comment]
> > Sorry you mean remove the bold ?
> 
> I meant to use the same printf function as in the actual help:
> 
> ```makefile
> help:
> 	@printf "\033[1mAvailable targets for minikube ${VERSION}\033[21m\n"
> 	@printf "\033[1m-------------------------------------\033[21m\n"
> 	@grep -h -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
> ```

Done. Thanks.

**Comment by afbjorklund** [comment]
The problem is that `\e` is **not** working, need to use `\033`

```
\e[1mAvailable targets for minikube v1.4.1\e[21m
\e[1m--------------------------------------\e[21m
```

**Comment by nanikjava** [comment]
> The problem is that `\e` is **not** working, need to use `\033`
> 
> ```
> \e[1mAvailable targets for minikube v1.4.1\e[21m
> \e[1m--------------------------------------\e[21m
> ```

Done. Really weird it's not working on your end with \e.

**Comment by afbjorklund** [comment]
> Really weird it's not working on your end with \e.

Probably related to older bash versions ?

```console
$ bash --version
GNU bash, version 4.3.48(1)-release (x86_64-pc-linux-gnu)
Copyright © 2013 Free Software Foundation, Inc.
Licens GPLv3+: GNU GPL version 3 eller senare <http://gnu.org/licenses/gpl.html>
```

Thanks for fixing!

/ok-to-test

**Comment by nanikjava** [comment]
> ```
> $ bash --version
> GNU bash, version 4.3.48(1)-release (x86_64-pc-linux-gnu)
> Copyright © 2013 Free Software Foundation, Inc.
> Licens GPLv3+: GNU GPL version 3 eller senare <http://gnu.org/licenses/gpl.html>
> ```
> 

Interesting. Following is my bash info

GNU bash, version 5.0.3(1)-release (x86_64-pc-linux-gnu)
Copyright (C) 2019 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

**Comment by afbjorklund** [comment]
Well, it seems that it does work in newer versions of `/bin/sh` - but not in any version of `/bin/bash` ?

```console
$ sh -c 'echo "\e[1mbold\e[21m"'
\e[1mbold\e[21m
$ docker run debian:latest sh -c 'echo "\e[1mbold\e[21m"'
bold
$ bash -c 'echo "\e[1mbold\e[21m"'
\e[1mbold\e[21m
$ docker run debian:latest bash -c 'echo "\e[1mbold\e[21m"'
\e[1mbold\e[21m
```

```console
$ docker run debian:latest dpkg --list dash
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version      Architecture Description
+++-==============-============-============-=================================
ii  dash           0.5.10.2-5   amd64        POSIX-compliant shell
$ docker run debian:latest bash --version
GNU bash, version 5.0.3(1)-release (x86_64-pc-linux-gnu)
Copyright (C) 2019 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

As in: it (echo \e) works OK with Ubuntu 18.04 LTS but does **not** work on the older Ubuntu 16.04 LTS

File: go.mod [hunk 4]
```diff
@@ -21,11 +21,14 @@ require (
 	github.com/docker/machine v0.7.1-0.20190718054102-a555e4f7a8f5 // version is 0.7.1 to pin to a555e4f7a8f5
 	github.com/elazarl/goproxy v0.0.0-20190421051319-9d40249d3c2f
 	github.com/elazarl/goproxy/ext v0.0.0-20190421051319-9d40249d3c2f // indirect
+	github.com/ghodss/yaml v1.0.0 // indirect
```
Comment (medyagh) [inline go.mod]: no need to change the go.mod if you pull from master it has it

**Comment by medyagh** [comment]
@nanikjava thanks for bringing this PR to my attention, we were actually just talking about this PR yesterday, that what happend to it ! I was looking for it

**Comment by medyagh** [comment]
this looks much better ! @nanikjava thank you so much for making this happen ! I like the colors
Screen Shot 2019-11-07 at 1 10 34 PM

**Pull request merged**